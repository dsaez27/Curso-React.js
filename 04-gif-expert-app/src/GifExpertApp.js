import React, { useState } from 'react';
import {AddCategory} from './Components/AddCategory';
import {GifGrid} from './Components/GifGrid';

export const GifExpertApp = ({ defaultCategories = []}) => {
	const [categories, setCategories] = useState(defaultCategories);

	//? const handleAdd = () => setCategories(cats => [...cats, 'HunterXHunter']);

	//* const categories = ['One punch', 'Samurai X', 'Dragon Ball'];

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />

			<ol>
				{
					categories.map((category) => (
						<GifGrid
							key={category}
							category={category}					
						/>
					))
				}
			</ol>

		</>
	)
}

