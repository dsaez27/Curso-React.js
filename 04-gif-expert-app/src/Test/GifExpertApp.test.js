import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el <GifExpertApp />', () => {
    
    

    test('Debe de mostrarse <GifExpertApp /> correctamente', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', () => {
        const category = ['One Punch', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(category.length);
    });
});