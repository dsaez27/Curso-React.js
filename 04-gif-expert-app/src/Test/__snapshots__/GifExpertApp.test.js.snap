// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp /> Debe de mostrarse <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
